energy-dao/src/external_sc_interactions/farm_interactions.rs

```diff
@@ -0,0 +1,252 @@
+multiversx_sc::imports!();
+
+use crate::{
+    common::errors::{
+        ERROR_BAD_PAYMENT_TOKENS, ERROR_EXTERNAL_CONTRACT_OUTPUT, ERROR_FARM_DOES_NOT_EXIST,
+        ERROR_UNBOND_TOO_SOON,
+    },
+    external_sc_interactions::farm_config::{UnstakeTokenAttributes, WrappedFarmTokenAttributes},
+};
+use common_structs::PaymentsVec;
+use locked_token_wrapper::wrapped_token;
+
+#[multiversx_sc::module]
+pub trait FarmInteractionsModule:
+    crate::external_sc_interactions::farm_config::FarmConfigModule
+    + crate::external_sc_interactions::farm_actions::FarmActionsModule
+    + crate::external_sc_interactions::locked_token_actions::LockedTokenModule
+    + crate::external_sc_interactions::fees_collector_interactions::FeesCollectorInteractionsModule
+    + lkmex_transfer::energy_transfer::EnergyTransferModule
+    + legacy_token_decode_module::LegacyTokenDecodeModule
+    + energy_query::EnergyQueryModule
+    + token_send::TokenSendModule
+    + utils::UtilsModule
+    + wrapped_token::WrappedTokenModule
+    + simple_lock::token_attributes::TokenAttributesModule
+    + multiversx_sc_modules::default_issue_callbacks::DefaultIssueCallbacksModule
+{
+    #[payable("*")]
+    #[endpoint(enterFarm)]
+    fn enter_farm_endpoint(&self, farm_address: ManagedAddress) -> EsdtTokenPayment {
+        let payment = self.call_value().single_esdt();
+        let mut farm_config_mapper = self.farm_config(&farm_address);
+        require!(!farm_config_mapper.is_empty(), ERROR_FARM_DOES_NOT_EXIST);
+        let farm_config = farm_config_mapper.get();
+        require!(
+            farm_config.farming_token_id == payment.token_identifier,
+            ERROR_BAD_PAYMENT_TOKENS
+        );
+        let mut enter_farm_payments = ManagedVec::new();
+        enter_farm_payments.push(payment);
+
+        let current_farm_position = EsdtTokenPayment::new(
+            farm_config.farm_token_id.clone(),
+            farm_config.farm_token_nonce,
+            farm_config.farm_staked_value.clone(),
+        );
+        let initial_total_farm_amount = current_farm_position.amount.clone();
+        if initial_total_farm_amount > 0 {
+            enter_farm_payments.push(current_farm_position);
+        }
```

Comment:
No need to clone here.
```suggestion
        let initial_total_farm_amount = &current_farm_position.amount;
        if initial_total_farm_amount > &0 {
            enter_farm_payments.push(current_farm_position);
        }
```

Reply:
I cannot do this, as after a few more lines from`enter_farm_payments.push(current_farm_position);` I use `initial_total_farm_amount` twice, and it is no longer available, as the variable from were it was borrowed does not exist anymore.